discovery/generator.go

```diff
@@ -0,0 +1,52 @@
+package discovery
+
+import (
+	"fmt"
+	"regexp"
+)
+
+func incString(s string) string {
+	if s == "" {
```

Comment:
All those `if`/`else if`/`else` could probably be replaced by a `switch` to make it more clear, with maybe a comment on each case explaining why?